hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java

```diff
@@ -70,6 +71,11 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         throwIfUnsuccessful(outcome.status());
     }
 
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        // nothing to do
```

Comment:
Question - why are we doing nothing here and in the `EthereumTransactionHandler` below instead of checking the op is the correct type?  I understand that the contractID may not exist.

Reply:
The handlers are dispatched based on the op type. In order for code in this class to be called, op must already be of type ContractCreate

Reply:
Makes sense but it seems we are checking the type in some of the handlers.